sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_document_retrieval/_document_retrieval.py

```diff
@@ -0,0 +1,342 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import math
+import operator
+from itertools import starmap
+from typing import Dict, List, TypedDict, Optional
+from azure.ai.evaluation._evaluators._common import EvaluatorBase
+from azure.ai.evaluation._exceptions import EvaluationException
+from typing_extensions import override, overload
+
+
+RetrievalGroundTruthDocument = TypedDict(
+    "RetrievalGroundTruthDocument", {"document_id": str, "query_relevance_label": int}
+)
+
+RetrievedDocument = TypedDict(
+    "RetrievedDocument", {"document_id": str, "relevance_score": float}
+)
+
+DocumentRetrievalMetrics = TypedDict(
+    "DocumentRetrievalMetrics",
+    {
+        "ndcg@3": float,
+        "ndcg@3_result": bool,
+        "xdcg@3": float,
+        "xdcg@3_result": bool,
+        "fidelity": float,
+        "fidelity_result": bool,
+        "top1_relevance": float,
+        "top1_relevance_result": bool,
+        "top3_max_relevance": float,
+        "top3_max_relevance_result": bool,
+        "holes": int,
+        "holes_result": bool,
+        "holes_ratio": float,
+        "holes_ratio_result": bool,
+        "total_retrieved_documents": int,
+        "total_groundtruth_documents": int,
+    },
+)
+
+
+class DocumentRetrievalEvaluator(EvaluatorBase):
+    """
```

Comment:
Pleae add more detailed doc string for the class. Do you use any existing evaluator for reference. We usually reference a sample too in doc string

Reply:
Updated and added a sample.  Let me know if you have other suggestions, particularly with the samples that I have added for this.